src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,793 @@
+      let currentDraggedDistanceX =
+        this.#lastDragEvent.clientX - this.#initialDragEvent.clientX;
+
+      if (this.#isRTLDirection) {
+        currentDraggedDistanceX = -currentDraggedDistanceX;
+      }
+
+      // Start direction inverts the increment
+      if (direction === "start") {
+        currentDraggedDistanceX = -currentDraggedDistanceX;
+      }
+
+      const newInlineSize = dialogRect.width + currentDraggedDistanceX;
+      const newRestrictedInlineSize = forceCSSMinMax(
+        newInlineSize,
+        this.#minInlineSize,
+        this.#maxInlineSize
+      );
+
+      // Do not apply resizes or translations if the control is at the minimum
+      // or maximum size
+      if (newRestrictedInlineSize === dialogRect.width) {
+        return;
+      }
+
+      // - - - - - - - - - - - - - DOM write operations - - - - - - - - - - - - -
+      // By resizing the start edge the control is translated to improve the UX
+      if (direction === "start") {
+        this.#draggedDistanceXForResize -= currentDraggedDistanceX;
+
+        setProperty(this.el, DIALOG_DRAGGED_X, this.#draggedDistanceXForResize);
+      }
+
+      setProperty(this.el, DIALOG_INLINE_SIZE, newRestrictedInlineSize);
+    }
+  } as const;
+
+  #resizeEdgesAndCornersDictionary: {
+    [key in ChDialogResizeElement]: (dialogRect: DOMRect) => void;
+  } = {
+    "block-start": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "start"),
+
+    "block-end": dialogRect =>
+      this.#resizeByDirectionDictionary.block(dialogRect, "end"),
+
+    "inline-start": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start"),
+
+    "inline-end": dialogRect =>
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end"),
+
+    "block-start-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-start-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "start");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    },
+
+    "block-end-inline-start": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "start");
+    },
+
+    "block-end-inline-end": dialogRect => {
+      this.#resizeByDirectionDictionary.block(dialogRect, "end");
+      this.#resizeByDirectionDictionary.inline(dialogRect, "end");
+    }
+  };
+
+  // Refs
+  #resizeLayer: HTMLDivElement;
+  #dialogRef: HTMLDialogElement;
+
+  @Element() el: HTMLChDialogElement;
+
+  @State() dragging = false;
+  @State() relativeDialog = false;
+  @State() resizing = false;
+
+  /**
+   * Specifies the drag behavior of the dialog.
+   * If `allowDrag === "header"`, a slot with the `"header"` name will be
+   * available to place the header content.
```

Comment:
This description does not achieve the behavior of the control. The header is only rendered if `showHeader === true`.